### PR TITLE
Feature: Search client, group, loanaccounts, savingsaccount, centers !!

### PR DESCRIPTION
**Pull request by therajanmaurya**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.
…ture added..
screen shot 2016-11-22 at 10 39 25 am

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/utils/Constants.java [hunk 1]
```diff
@@ -41,6 +41,17 @@
...
 
+    /**
+     * Search Entity
+     */
+    public static final String SEARCH_ENTITY_CLIENT = "CLIENT";
+    public static final String SEARCH_ENTITY_GROUP = "GROUP";
+    public static final String SEARCH_ENTITY_LOAN = "LOAN";
+    public static final String SEARCH_ENTITY_SAVING = "SAVING";
+    public static final String SEARCH_ENTITY_CENTER = "CENTER";
+
```
Comment (ishan1604) [inline mifosng-android/src/main/java/com/mifos/utils/Constants.java]: Too many new lines 😞  For no reason

**Comment by ishan1604** [comment]
Excellent Job abstracting out the SearchAdapter 👍  

Quite an important feature for discovering a lot of stuff. Kudos !! 

@edcable probably one of the things people would absolutely love 😃

File: mifosng-android/src/main/java/com/mifos/utils/Constants.java (hunk 1 above)
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/utils/Constants.java]: Sorry 😬, It's done !!

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/CentersActivity.java [hunk 2]
```diff
@@ -17,18 +17,26 @@
...
 
+    private int centerId = 0;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_toolbar_container);
         showBackButton();
-        replaceFragment(new CenterListFragment(), false, R.id.container);
+        centerId = getIntent().getExtras().getInt(Constants.CENTER_ID);
```
Comment (mayank-kgp) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/CentersActivity.java]: @therajanmaurya  This line is causing app crash as there is no extras if we navigate from dashboardactivity.java (home page from app).
Can I solve it by opening an issue and sending a Quick PR by putting a condition if getExtras() is not null.
Or do you want anything to do with changing in this pr?

Comment (mayank-kgp) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/CentersActivity.java]: @therajanmaurya  Please see into it. @tarun0 has sent PR related to this. 
https://github.com/openMF/android-client/pull/458

**Pull request merged**